### PR TITLE
tests: use ${CFLAGS}

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
Heh, not much point having a strict `$CFLAGS` variable if it's not used in (most) of the tests!

**Comment by gperciva** [comment]
Not certain if you'll like splitting up the getopt test, but it works.

**Comment by cperciva** [comment]
I don't really want to waste time working around compiler-is-stupid warnings when they only related to test code.  Is this all we need?

**Comment by gperciva** [comment]
Yes, that's why this PR passed the Travis-CI tests...?

I'm still working on parsenum stuff, but that's not part of this particular PR.

**Comment by cperciva** [comment]
I wasn't sure if you had all the warnings turned on which you wanted. :-)

**Comment by gperciva** [comment]
Ah.  Yeah, that's the full set of warnings that we can use with clang-3.5.  (note that this commit doesn't modify the actual `CFLAGS` at all; it merely tells the `Makefile`s to use them.  The sha256 test already did so, probably because you needed them for `posix-cflags.sh`.)

As per the commit message in fe75d3acb359006b79ee2a37a8636bd56b44af3f (the one that added the strict tests to Travis-CI), once Travis-CI uses a newer clang, we'll need to add in `-Wno-reserved-id-macro`.

File: tests/getopt/main.c [hunk 1]
```diff
@@ -11,15 +11,12 @@ usage(void)
 	exit(1);
 }
 
-int
-main(int argc, char * argv[])
+static void
+parse_first(int argc, char * argv[])
```
Comment (cperciva) [inline tests/getopt/main.c]: I'd call these `pass1` and `pass2`.

File: tests/getopt/main.c [hunk 2]
```diff
@@ -81,5 +87,18 @@ main(int argc, char * argv[])
...
+}
+
+int
+main(int argc, char * argv[])
+{
+
+	/* Process arguments. */
+	parse_first(argc, argv);
+
+	/* We want to make sure that reset works. */
```
Comment (cperciva) [inline tests/getopt/main.c]: `/* Reset getopt state and process arguments again. */`

Comment (gperciva) [inline tests/getopt/main.c]: For the record, that comment was in the original file.

**Comment by gperciva** [comment]
Updated with `pass1()`, `pass2()`, and changing the `Reset getopt...` comment.  I rebased those changes with the previous commit, since they seemed simple enough.

File: tests/getopt/main.c (hunk 2 above)
Comment (cperciva) [inline tests/getopt/main.c]: Yeah, but it made far more sense in its original context. :-)

**Pull request merged**